appium/lib/screen-object.rb

```diff
@@ -37,127 +36,161 @@ def driver
-        false
-      end
-    end
+    driver.send_keys(:enter)
   end
 
-  def scroll_down_click(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          driver.find_element(locator,locator_value).click
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollStart,:end_x =>0,:end_y =>scrollEnd,:touchCount => 2,:duration => 0)
-        false
-      end
-    end
+  def gesture(arg)
+    Appium::TouchAction.new(driver).swipe(start_x: arg[0], start_y: arg[1], end_x: arg[2], end_y: arg[3], duration: arg[4]).perform
+  rescue RuntimeError => e
+    raise("Error during gesture \n Error Details: #{e}")
   end
 
-  def scroll_up_find(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollEnd,:end_x =>0,:end_y =>scrollStart,:touchCount => 2,:duration => 0)
-        false
-      end
-    end
+  def scroll(direction = :down, duration = 1000)
+    size = driver.window_size
+    x = size.width / 2
+    y = size.height / 2
+    loc = case direction
+          when :up then    [x, y * 0.5, x, (y + (y * 0.5)), duration]
+          when :down then  [x, y, x, y * 0.5, duration]
+          when :left then  [x * 0.6, y, x * 0.3, y, duration]
+          when :right then [x * 0.3, y, x * 0.6, y, duration]
+          else
+            raise('Only upwards, downwards, leftwards and rightwards scrolling are supported')
+          end
+    gesture(loc)
   end
 
+  def scroll_down
+    scroll(:down)
+  end
 
-  def scroll_up_click(locator,locator_value,num_loop = 15)
-    scr = driver.window_size
-    screenHeightStart = (scr.height) * 0.5
-    scrollStart = screenHeightStart.to_i
-    screenHeightEnd = (scr.height) * 0.2
-    scrollEnd = screenHeightEnd.to_i
-    for i in 0..num_loop
-      begin
-        if (driver.find_element(locator,locator_value).displayed?)
-          driver.find_element(locator,locator_value).click
-          break
-        end
-      rescue
-        driver.swipe(:start_x => 0,:start_y => scrollEnd,:end_x =>0,:end_y =>scrollStart,:touchCount => 2,:duration => 0)
-        false
-      end
+  def scroll_up
+    scroll(:up)
+  end
+
+  def swipe_left
+    scroll(:left)
+  end
+
+  def swipe_right
+    scroll(:right)
+  end
+
+  def swipe_element(locator, direction = :down, duration = 1000)
```

Comment:
This can probably be moved to accessors.rb under Accessors module > element class as a dynamic method.

`define_method("#{name}_swipe") do .. end`

That way you won't have to pass locator, you can directly use the swipe method using the screen object's name, example: `unlock_button_swipe`